ApplicationService/connection_pool.py

```diff
@@ -0,0 +1,66 @@
+import os
+import psycopg2
+import threading
+from ApplicationService.identity import identity
+
+
+class connection_pool:
+    def get_connection(self, identifier: identity):
+        raise NotImplementedError()
+
+    def get_cursor(self, identifier: identity):
+        raise NotImplementedError()
+
+    def refund(self, identifier: identity):
+        raise NotImplementedError()
+
+class connection:
+    def cursor(self):
+        raise NotImplementedError
+
+
+class postgresql_connection_pool(connection_pool):
+    def __init__(self, createConnection=None, max_connections=1, condition=None):
+        self.create_connection = createConnection or psycopg2_create_connection
+        self.connections: dict[int, tuple] = dict()  # tuple[0] are connections and tuple[1] are conn's cursor
+        self.max_connections = max_connections
+        self.free_connections = []
+        self.condition = condition() if condition else threading.Condition()
+
+    def get_connection(self, identifier: identity):
+        id_conn = identifier.value()
+        with self.condition:
+            if id_conn not in self.connections.keys():
+                while True:
+                    if self.free_connections:
+                        conn = self.free_connections.pop()
+                        self.connections[id_conn] = (conn, conn.cursor())
+                        break
+
+                    elif len(self.connections) < self.max_connections:
```

Comment:
change name to self.alocatedConnections or something like that, is more semantic, dont you think?